x-pack/solutions/security/plugins/security_solution/public/management/pages/endpoint_hosts/view/hooks/insights/use_fetch_ongoing_tasks.ts

```diff
@@ -29,57 +30,107 @@ export const useFetchLatestScan = ({
 }: UseFetchOngoingScansConfig) => {
   const { http } = useKibana().services;
   const toasts = useToasts();
+  const defendInsightsPolicyResponseFailureEnabled = useIsExperimentalFeatureEnabled(
+    'defendInsightsPolicyResponseFailure'
+  );
 
-  return useQuery<
-    DefendInsightsResponse | undefined,
-    { body?: { error: string } },
-    DefendInsightsResponse | undefined
-  >(
-    [`fetchOngoingTasks-${endpointId}`],
+  return useQuery<{ hasRunning: boolean }, { body?: { error: string } }, { hasRunning: boolean }>(
+    [`fetchOngoingTasks-${endpointId}`, defendInsightsPolicyResponseFailureEnabled],
     async () => {
       try {
-        const response = await http.get<{ data: DefendInsightsResponse[] }>(DEFEND_INSIGHTS, {
-          version: API_VERSIONS.internal.v1,
-          query: {
-            endpoint_ids: [endpointId],
-            size: 1,
-          },
-        });
+        const makeSingleQuery = async (): Promise<DefendInsightsResponse[]> => {
+          const response = await http.get<{ data: DefendInsightsResponse[] }>(DEFEND_INSIGHTS, {
+            version: API_VERSIONS.internal.v1,
+            query: {
+              endpoint_ids: [endpointId],
+              size: 1,
+            },
+          });
+          return response.data;
+        };
 
-        const defendInsight = response.data[0];
-        if (!defendInsight) {
-          // no previous scan record - treat as 0 expected insights
-          onSuccess(0);
-          return undefined;
-        }
+        const makeMultiTypeQueries = async (): Promise<DefendInsightsResponse[]> => {
+          const insightTypes = ['incompatible_antivirus', 'policy_response_failure'];
 
-        const status = defendInsight.status;
+          const queries = insightTypes.map((type) =>
```

Comment:
This works but I'm a bit concerned about how this will scale as we add more insight types. Making 1 call per insight type every polling interval can grow out of control quickly. Would it be better to skip the type filter and just size: `insightType.length`?